core/domain/draft_upgrade_services.py

```diff
@@ -129,24 +129,29 @@ def _convert_html_in_draft_change_list(
             list(ExplorationChange). The converted draft_change_list.
         """
         for i, change in enumerate(draft_change_list):
-            new_value = None
             if not change.cmd == exp_domain.CMD_EDIT_STATE_PROPERTY:
                 continue
             # The change object has the key 'new_value' only if the
             # cmd is 'CMD_EDIT_STATE_PROPERTY' or
             # 'CMD_EDIT_EXPLORATION_PROPERTY'.
             new_value = change.new_value
             if change.property_name == exp_domain.STATE_PROPERTY_CONTENT:
-                # Ruling out the possibility of any other type for mypy
-                # type checking.
+                # Ruling out the possibility of any other type for MyPy
+                # type checking, because here 'new_value' is a value of
+                # attribute that was dynamically defined on change_domain
+                # and that attribute has str type. Thus to convert the
```

Comment:
I think that considering new_value to be a string is an error, and we have a way forward to address this, but let me know if that's not the case.

Ditto below (several places).

Reply:
This issue is fixed under PR #15958, but PR 15958 can only be available to review when this PR is merged because #15958 is built on top of this PR and shows this PR's changes as well.

Reply:
Hope it sounds good!!

Reply:
OK, yup, I think that's fine -- thanks!